networkx/algorithms/isomorphism/vf2pp_helpers/feasibility.py

```diff
@@ -0,0 +1,181 @@
+    """
+    G1, G2 = graph_params.G1, graph_params.G2
+    if G1.number_of_edges(node1, node1) != G2.number_of_edges(node2, node2):
+        return False
+
+    if _cut_PT(node1, node2, graph_params, state_params):
+        return False
+
+    if isinstance(G1, nx.MultiGraph):
+        if not _consistent_PT(node1, node2, graph_params, state_params):
+            return False
+
+    return True
+
+
+def _cut_PT(u, v, graph_params, state_params):
+    """Implements the cutting rules for the ISO problem.
+
+    Parameters
+    ----------
+    u, v: Graph node
+        The two candidate nodes being examined.
+
+    graph_params: namedtuple
+        Contains all the Graph-related parameters:
+
+        G1,G2: NetworkX Graph or MultiGraph instances.
+            The two graphs to check for isomorphism or monomorphism
+
+        G1_labels,G2_labels: dict
+            The label of every node in G1 and G2 respectively
+
+    state_params: namedtuple
+        Contains all the State-related parameters:
+
+        mapping: dict
+            The mapping as extended so far. Maps nodes of G1 to nodes of G2
+
+        reverse_mapping: dict
+            The reverse mapping as extended so far. Maps nodes from G2 to nodes of G1. It's basically "mapping" reversed
+
+        T1, T2: set
+            Ti contains uncovered neighbors of covered nodes from Gi, i.e. nodes that are not in the mapping, but are
+            neighbors of nodes that are.
+
+        T1_out, T2_out: set
+            Ti_out contains all the nodes from Gi, that are neither in the mapping nor in Ti
+
+    Returns
+    -------
+    True if we should prune this branch, i.e. the node pair failed the cutting checks. False otherwise.
+    """
+    G1, G2, G1_labels, G2_labels, _, _, _ = graph_params
+    _, _, T1, T1_out, T2, T2_out = state_params
+
+    u_labels_neighbors = nx.utils.groups({n1: G1_labels[n1] for n1 in G1[u]})
+    v_labels_neighbors = nx.utils.groups({n2: G2_labels[n2] for n2 in G2[v]})
+
+    # if the neighbors of u, do not have the same labels as those of v, NOT feasible.
+    if set(u_labels_neighbors.keys()) != set(v_labels_neighbors.keys()):
+        return True
+
+    for label, G1_nbh in u_labels_neighbors.items():
+        G2_nbh = v_labels_neighbors[label]
+
+        if isinstance(G1, nx.MultiGraph):
+            # Check for every neighbor in the neighborhood, if u-nbr1 has same edges as v-nbr2
+            u_nbrs_edges = sorted(G1.number_of_edges(u, x) for x in G1_nbh)
+            v_nbrs_edges = sorted(G2.number_of_edges(v, x) for x in G2_nbh)
+            if any(
+                u_nbr_edges != v_nbr_edges
+                for u_nbr_edges, v_nbr_edges in zip(u_nbrs_edges, v_nbrs_edges)
+            ):
+                return True
+
+        if len(T1.intersection(G1_nbh)) != len(T2.intersection(G2_nbh)) or len(
+            T1_out.intersection(G1_nbh)
+        ) != len(T2_out.intersection(G2_nbh)):
+            return True
```

Comment:
```suggestion
        if len(T1.intersection(G1_nbh)) != len(T2.intersection(G2_nbh)):
            return True
        if len(T1_out.intersection(G1_nbh)) != len(T2_out.intersection(G2_nbh)):
            return True
```
The ugly black-caused spacing is unfortunate. This might make it more readable. Your choice...